### DISCUSSION TITLE
hue linearity vs delta uniformity

### DISCUSSION THREAD

**Discussion by swamidass** [coloria-dev/coloria#198] [Q&A]
It seems that hue linearity is in tension with delta uniformity, and color spaces have to make a trade off between the two.

Are there any standard approaches to resolving this tension? Perhaps, for example, we could use a hue linear color space, but use a non-linear delta function in this space to govern how to mix colors and move around the space with perceptual uniformity. It seems this might be possible by directly using the perceptual data in an efficient way.

I'm asking here because @nschloe seems more informed about color science than many, and this package embeds the empirical data.

**Comment by nschloe** [comment]
> Are there any standard approaches to resolving this tension?

No, and there can't be. Most new articles often make a "creative" decision on which data sets they pick as reference, and the optimize against that. I don't endorse this practice.

> Perhaps, for example, we could use a hue linear color space, but use a non-linear delta function in this space to govern how to mix colors and move around the space with perceptual uniformity. It seems this might be possible by directly using the perceptual data in an efficient way.

Depends on what you want to do, right? What's your goal exactly? Construct a hue-linear color space? Alright, just take all hue-linearity results and optimize against those. Want a perceptually uniform color space? Okay, optimize against the respective experiments.

I don't really understand the color space you're suggesting, and my suspicion is you don't either. :smiley_cat: One thing to note is that hue-linearity is directly related to "how you mix colors". If you mix two blues, you want blue again, not purple. Hue linearity gives you just that.

The problem of perptual uniformity is more difficult to treat.
